Documentation/Browser/BrowsingContextsAndNavigables.md

```diff
@@ -0,0 +1,128 @@
+in some specification documents) defined extra state that needs to be associated with the realm.
+In LibWeb, the Host Defined slot holds an object that has the HTML Environment Settings Object for each realm,
+as well as all the prototypes, constructors, and namespaces that need to be exposed on the Global Object
+for Web APIs. On top of the Realm abstraction, ECMAScript uses the Execution Context to model the state
+of execution of one particular script or module. On top of the Realm it refers to, the Execution Context
+has information about the current function, the script or module that the current code block belongs to,
+additional Environment Records required to access names in the current scope, any running generator state,
+and most importantly to the thread analogy, the state needed to suspend and resume execution of that script.
+Throughout the ECMAScript specification, the text refers to a stack of execution contexts, with the uppermost
+one always being the [*running execution context*](https://tc39.es/ecma262/#running-execution-context).
+As with Environment Records, Realms and Execution contexts are not visible to running JavaScript code.
+
+The final missing piece for the JavaScript execution model is how these stacks of Execution Contexts
+are actually scheduled to run by the ECMAScript implementation. In the most common case, this means directly
+mapping the ECMAScript model to the earlier native concepts of threads and processes in a way that
+allows for flexibility in the implementation strategies. The last thing that the specification authors want
+to do is constrain implementations so much that innovation and experimentation becomes impossible.
+The method for this mapping is the two related specification objects [*Agents*](https://tc39.es/ecma262/#sec-agents)
+and [*Agent Clusters*](https://tc39.es/ecma262/#sec-agent-clusters). The Execution Context stack mentioned
+above actually belongs to an Agent, which holds said stack, a set of metadata about the memory model,
+and a shared reference to an [*executing thread*](https://tc39.es/ecma262/#executing-thread). To re-iterate,
+an Agent holds a stack of Execution Contexts, with the topmost entry being the running execution context.
+Each Execution Context holds a Realm and a specific script's context, including the current function and
+any state required to pause and resume the execution for that context. The Realm holds the Global
+Object for the Execution Context, and any ECMAScript or host-specific intrinsics required to create the
+desired environment for code to run in. More loosely, an Agent is a specification artefact that somewhat
+maps the execution of a JavaScript script or module to a native thread of execution. But the specification
+does so in a way that allows a host/embedder to choose to switch out which Agent is currently executing
+its running execution context on that native thread.
```

Comment:
I don't think this was mentioned before but is certainly relevant information: more than one Realm can exist in any given Agent. Current ECMAScript spec would only ever create one, but I'm sure the Web does that somewhere, and https://github.com/tc39/proposal-shadowrealm will directly expose this ability to user code.

Reply:
So yes, it seems like I should mention that we can have multiple realms per agent :). I thought I'd mentioned it but seems not.